src/cascadia/TerminalApp/SuggestionsControl.cpp

```diff
@@ -281,11 +282,91 @@ namespace winrt::TerminalApp::implementation
         {
             if (const auto actionPaletteItem{ filteredCommand.Item().try_as<winrt::TerminalApp::ActionPaletteItem>() })
             {
-                PreviewAction.raise(*this, actionPaletteItem.Command());
+                const auto& cmd = actionPaletteItem.Command();
+                PreviewAction.raise(*this, cmd);
+
+                const auto description{ cmd.Description() };
+
+                if (const auto& selected{ SelectedItem() })
+                {
+                    selected.SetValue(Automation::AutomationProperties::FullDescriptionProperty(), winrt::box_value(description));
+                }
+
+                if (!description.empty())
+                {
+                    _openTooltip(cmd);
+                }
+                else
+                {
+                    // If there's no description, then just close the tooltip.
+                    _descriptionsView().Visibility(Visibility::Collapsed);
+                    _descriptionsBackdrop().Visibility(Visibility::Collapsed);
+                    _recalculateTopMargin();
+                }
             }
         }
     }
 
+    winrt::fire_and_forget SuggestionsControl::_openTooltip(Command cmd)
+    {
+        const auto description{ cmd.Description() };
+        if (description.empty())
+        {
+            co_return;
+        }
+
+        // If it's already open, then just re-target it and update
+        // the content immediately. Otherwise:
+        // wait a bit before opening it.
+        if (_descriptionsView().Visibility() != Visibility::Visible)
+        {
+            co_await winrt::resume_after(200ms);
+            co_await wil::resume_foreground(Dispatcher());
```

Comment:
Do we need to recheck if some state changed in the meantime? What if `_openTooltip` gets called a 2nd time while this time is still running?